crates/miden-lib/src/account/interface/component.rs

```diff
@@ -0,0 +1,254 @@
+    /// ```
+    ///
+    /// Example script for the [`AccountComponentInterface::BasicFungibleFaucet`] with one note:
+    ///
+    /// ```masm
+    ///     push.{note information}
+    ///     
+    ///     push.{asset amount}
+    ///     call.::miden::contracts::faucets::basic_fungible::distribute dropw dropw drop
+    /// ```
+    ///
+    /// # Errors:
+    /// Returns an error if:
+    /// - the interface does not support the generation of the standard `send_note` procedure.
+    /// - the sender of the note isn't the account for which the script is being built.
+    /// - the note created by the faucet doesn't contain exactly one asset.
+    /// - a faucet tries to distribute an asset with a different faucet ID.
+    pub(crate) fn send_note_body(
+        &self,
+        sender_account_id: AccountId,
+        notes: &[PartialNote],
+    ) -> Result<String, AccountInterfaceError> {
+        let mut body = String::new();
+
+        for partial_note in notes {
+            if partial_note.metadata().sender() != sender_account_id {
+                return Err(AccountInterfaceError::InvalidSenderAccount(
+                    partial_note.metadata().sender(),
+                ));
+            }
+
+            body.push_str(&format!(
+                "push.{recipient}
+                push.{execution_hint}
+                push.{note_type}
+                push.{aux}
+                push.{tag}\n",
+                recipient = word_to_masm_push_string(&partial_note.recipient_digest()),
+                note_type = Felt::from(partial_note.metadata().note_type()),
+                execution_hint = Felt::from(partial_note.metadata().execution_hint()),
+                aux = partial_note.metadata().aux(),
+                tag = Felt::from(partial_note.metadata().tag()),
+            ));
+            // stack => [tag, aux, note_type, execution_hint, RECIPIENT]
+
+            match self {
+                AccountComponentInterface::BasicFungibleFaucet => {
+                    if partial_note.assets().num_assets() != 1 {
+                        return Err(AccountInterfaceError::FaucetNoteWithoutAsset);
+                    }
+
+                    // SAFETY: We checked that the note contains exactly one asset
+                    let asset =
+                        partial_note.assets().iter().next().expect("note should contain an asset");
+
+                    if asset.faucet_id_prefix() != sender_account_id.prefix() {
+                        return Err(AccountInterfaceError::IssuanceFaucetMismatch(
+                            asset.faucet_id_prefix(),
+                        ));
+                    }
+
+                    body.push_str(&format!(
+                        "push.{amount} 
+                        call.::miden::contracts::faucets::basic_fungible::distribute dropw dropw drop\n",
+                        amount = asset.unwrap_fungible().amount()
+                    ));
+                    // stack => [note_idx]
```

Comment:
Is this correct? Why do we leave `note_idx` on the stack? (we don't do this in case of `BasicWallet` on line 227 below).